packages/recurring-payments-job/src/__tests__/recurring-payments-processor.spec.js

```diff
@@ -337,10 +406,48 @@ describe('recurring-payments-processor', () => {
       await processRecurringPayments()
       expect(sendPayment.mock.calls).toEqual(expectedData)
     })
+
+    it('gets the payment status for each one', async () => {
+      const mockGetDueRecurringPayments = []
+      const agreementIds = []
+      for (let i = 0; i < count; i++) {
+        const agreementId = Symbol(`agreementId${1}`)
+        agreementIds.push(agreementId)
+        mockGetDueRecurringPayments.push(getMockDueRecurringPayment(i, agreementId))
+      }
+      salesApi.getDueRecurringPayments.mockReturnValueOnce(mockGetDueRecurringPayments)
+
+      const permits = []
+      for (let i = 0; i < count; i++) {
+        permits.push(Symbol(`permit${i}`))
+      }
+
+      permits.forEach((permit, i) => {
+        salesApi.preparePermissionDataForRenewal.mockReturnValueOnce({
+          licensee: { countryCode: 'GB-ENG' }
+        })
+
+        salesApi.createTransaction.mockReturnValueOnce({
+          cost: i,
+          id: permit
+        })
+      })
+
+      const expectedData = []
+      permits.forEach(() => {
+        expectedData.push(['test-payment-id'])
+      })
+
+      const mockPaymentResponse = { payment_id: 'test-payment-id' }
+      sendPayment.mockResolvedValue(mockPaymentResponse)
+
+      await processRecurringPayments()
+      expect(getPaymentStatus.mock.calls).toEqual(expectedData)
```

Comment:
Rather than directly interacting with the mock, use `toHaveBeenCalledWith(expectedData)`. Also, I think we need to have unique payment ids rather than using 'test-payment-id' for every one.